api/service/legacysync/helpers.go

```diff
@@ -0,0 +1,99 @@
+package legacysync
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/harmony-one/harmony/api/service/legacysync/downloader"
+	"github.com/harmony-one/harmony/internal/utils"
+	"github.com/harmony-one/harmony/p2p"
+	"github.com/pkg/errors"
+)
+
+// getMaxPeerHeight gets the maximum blockchain heights from peers
+func getMaxPeerHeight(syncConfig *SyncConfig) uint64 {
+	maxHeight := uint64(math.MaxUint64)
+	var (
+		wg   sync.WaitGroup
+		lock sync.Mutex
+	)
+
+	syncConfig.ForEachPeer(func(peerConfig *SyncPeerConfig) (brk bool) {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			//debug
+			// utils.Logger().Debug().Bool("isBeacon", isBeacon).Str("peerIP", peerConfig.ip).Str("peerPort", peerConfig.port).Msg("[Sync]getMaxPeerHeight")
+			response, err := peerConfig.client.GetBlockChainHeight()
+			if err != nil {
+				utils.Logger().Warn().Err(err).Str("peerIP", peerConfig.ip).Str("peerPort", peerConfig.port).Msg("[Sync]GetBlockChainHeight failed")
+				syncConfig.RemovePeer(peerConfig)
+				return
+			}
+			utils.Logger().Info().Str("peerIP", peerConfig.ip).Uint64("blockHeight", response.BlockHeight).
+				Msg("[SYNC] getMaxPeerHeight")
+
+			lock.Lock()
+			if response != nil {
+				if response.BlockHeight < math.MaxUint32 { // That's enough for decades.
+					if maxHeight == uint64(math.MaxUint64) || maxHeight < response.BlockHeight {
+						maxHeight = response.BlockHeight
+					}
+				}
+			}
+			lock.Unlock()
+		}()
+		return
+	})
+	wg.Wait()
+	return maxHeight
+}
+
+func createSyncConfig(syncConfig *SyncConfig, peers []p2p.Peer, isBeacon bool) (*SyncConfig, error) {
```

Comment:
Please refactor this so the code can be reused and not just a copy of the original code.